### PR TITLE
Maintenance and bugfix

### PR DESCRIPTION
**Pull request by guggero**
~Depends on https://github.com/lightninglabs/pool/pull/224.~
Replaces #185.

- Bumps lnd to `v0.12.1-beta`, Loop to `v0.11.4-beta` and Pool to `v0.4.4-alpha`.
- Bump golang Docker base image version to `v1.16.0`.
- Fix bug in integrated daemon mode not working with recent remote mode support.
- Update LiT version to `v0.4.1-alpha`

### CODE REVIEW

**Review by jamaljsr — APPROVED** [review]
Tested on regtest and testnet. LGTM 👍 

Thanks for fixing the gRPC issue. 

I noticed that after running `make install`, the `go.sum` file is modified. That has never happened before. Is this ok?

**Comment by guggero** [comment]
Fixed the diff in `go.sum` that was produced when running `make go-install-cli`. Thanks for catching that @jamaljsr!

File: go.sum [hunk 1]
```diff
@@ -245,9 +246,12 @@ github.com/lightningnetwork/lightning-onion v1.0.2-0.20200501022730-3c8c8d0b89ea
 github.com/lightningnetwork/lightning-onion v1.0.2-0.20200501022730-3c8c8d0b89ea h1:oCj48NQ8u7Vz+MmzHqt0db6mxcFZo3Ho7M5gCJauY/k=
 github.com/lightningnetwork/lightning-onion v1.0.2-0.20200501022730-3c8c8d0b89ea/go.mod h1:rigfi6Af/KqsF7Za0hOgcyq2PNH4AN70AaMRxcJkff4=
 github.com/lightningnetwork/lightning-onion v1.0.2-0.20200501022730-3c8c8d0b89ea/go.mod h1:rigfi6Af/KqsF7Za0hOgcyq2PNH4AN70AaMRxcJkff4=
-github.com/lightningnetwork/lnd v0.12.0-beta h1:m4Whrt0dvlnVD6OQT8kDd2utulnoZEcrq8BJlEPSiiY=
-github.com/lightningnetwork/lnd v0.12.0-beta/go.mod h1:2GyP1IG1kXV5Af/LOCxnXfux1OP3fAGr8zptS5PB2YI=
-github.com/lightningnetwork/lnd/cert v1.0.2 h1:g2rEu+sM2Uyz0bpfuvwri/ks6R/26H5iY1NcGbpDJ+c=
+github.com/lightningnetwork/lnd v0.11.0-beta/go.mod h1:CzArvT7NFDLhVyW06+NJWSuWFmE6Ea+AjjA3txUBqTM=
```
Comment (Roasbeef) [inline go.sum:249]: Does a `go mod tidy` get rid of these? Seemed to be able to get a slimmer dep set with my other PR when it used the replace method.

**Review by Roasbeef — APPROVED** [review]
LGTM 🐊

**Pull request merged**